reposerver/repository/repository_test.go

```diff
@@ -1587,3 +1587,19 @@ func Test_getHelmDependencyRepos(t *testing.T) {
 	assert.Equal(t, repos[0].Repo, repo1)
 	assert.Equal(t, repos[1].Repo, repo2)
 }
+
+func TestResolveRevision(t *testing.T) {
+
+	service := newService(".")
+	repo := &argoappv1.Repository{Repo: "https://github.com/argoproj/argo-cd"}
+	app := &argoappv1.Application{}
+	resolveRevisionResponse, err := service.ResolveRevision(context.Background(), &apiclient.ResolveRevisionRequest{
+		Repo:              repo,
+		App:               app,
+		AmbiguousRevision: "HEAD",
+	})
+
+	assert.NotNil(t, resolveRevisionResponse.Revision)
+	assert.Nil(t, err)
+
+}
```

Comment:
Maybe we can make this test a little more specific? For example, use a tag with a known commit SHA as `AmbiguousRevision` and check whether the response is what we expect and call it with a commit SHA to see if the result is the same.

Also, we should test for negative/error cases (e.g. unresolveable revision, unknown repository, etc)

Reply:
Makes sense. I will update the test case and add negative test cases as well.